bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostAsterism.scala

```diff
@@ -87,242 +83,132 @@ object GhostAsterism {
+      case DualTarget(t1,t2)  => interpolateCoords(t1.coordinates(when), t2.coordinates(when))
+      case TargetPlusSky(t,s) => interpolateCoords(t.coordinates(when), Some(s))
+      case SkyPlusTarget(s,t) => interpolateCoords(Some(s), t.coordinates(when))
+    }
   }
 
-  /** The high resolution asterism comes in two flavors, normal high resolution
-    * mode and precision radial velocity (PRV) mode.
-    */
-  sealed trait HighResMode extends Product with Serializable
-
-  object HighResMode {
+  object GhostStandardResTargets {
+    def interpolateCoords(c1Opt: Option[Coordinates], c2Opt: Option[Coordinates]): Option[Coordinates] = for {
+      c1 <- c1Opt
+      c2 <- c2Opt
+    } yield c1.interpolate(c2, 0.5)
 
-    /** Normal high res mode can use bright or faint binning options. (Prv is
-      * always bright).
-      */
-    case object Normal extends HighResMode
-
-    /** Prv can be set up with a fiber agitator. The simultaneous calibration
-      * lamp can default to an appropriate value for the target magnitude, or
-      * can be explicitly set to a particular duration.
-      *
-      * (TODO: Do we need to be able to turn off the lamp altogether?  Perhaps
-      * that is the same as setting the duration to 0.)
-      */
-    final case class Prv(agitator: FiberAgitatorState, calDuration: Option[Duration]) extends HighResMode
+    case class SingleTarget(target: GhostTarget) extends GhostStandardResTargets
+    case class DualTarget(target1: GhostTarget, target2: GhostTarget) extends GhostStandardResTargets
+    case class TargetPlusSky(target: GhostTarget, sky: Coordinates) extends GhostStandardResTargets
+    case class SkyPlusTarget(sky: Coordinates, target: GhostTarget) extends GhostStandardResTargets
   }
 
-  /** High resolution GHOST asterism.
+
+  /** High resolution mode.
+    *
+    * It has a calibration duration (of type Duration), which should be determined
+    * by exposure time.
     *
     * The target is always observed using the high resolution IFU1.  The sky
     * coordinates are observed using the sky fibers of IFU2, not SRIFU2. The
     * guide fibers will be used by default because the target must be bright,
     * but can be explicitly turned off.
     */
-  final case class HighRes(
-                     mode:        HighResMode,
-                     ghostTarget: GhostTarget,
-                     sky:         Coordinates) extends GhostAsterism {
+  sealed case class HighResolution(ghostTarget: GhostTarget,
+                                   sky: Coordinates,
+                                   base: Option[Coordinates]) extends GhostAsterism {
 
     override def allSpTargets: NonEmptyList[SPTarget] =
       NonEmptyList(ghostTarget.spTarget)
 
     /** Defines the base position to be the same as the target position. */
     override def basePosition(when: Option[Instant]): Option[Coordinates] =
-      ghostTarget.coordinates(when)
+      base orElse ghostTarget.coordinates(when)
```

Comment:
I wonder if `HighResolution` really will always have a sky position.  Certainly it was created that way in the previous version.  It isn't clear to me whether the sky position is in the second IFU or not despite, or maybe because of, the comment I wrote above the `HighResolution` class.  Let's check with Steve.  If it really is assigned to the second IFU, we should probably interpolate the coordinates as you did with the standard res target options.

Reply:
Sure thing. I asked him about the sky position earlier, but I just asked about the interpolation.

Reply:
From Steve:

"For HR or SR single target mode, the sky fibers should not used to pick the base position.  The default base position will be the same as the object IFU and the sky fibers will be at TBD default position."

I'll make those changes.

Reply:
I'm still confused about this. There are sky fibers at fixed positions, if I understand it correctly, and there are observations of sky positions at given coordinates? If you have one IFU at a given set of coordinates and the other at a different set of coordinates, why wouldn't the base position be in the middle.  If the sky is at a fixed position why do you supply coordinates?